### PR TITLE
Added Map of Code Schools #272 

### PR DESCRIPTION
**Pull request by hd719**
Hello @rickr 

I added the map of the codeschools (issue #272). 

I also could not add all the coordinates for all the coding bootcamps in the `config/code_schools.yml` file, but I add some just to test out the feature.

If I am missing anything please do let me know! 

screen shot 2017-05-01 at 9 55 07 am

-Thank you.

### CODE REVIEW

File: app/controllers/code_schools_controller.rb [hunk 1]
```diff
@@ -7,4 +7,20 @@ def index
...
+    code_schools = CodeSchools.new.all
+    @lat_longs = []
+
+    puts "first code school #{code_schools[0]}"
+
+    code_schools.each do |hash| 
+      if hash.key?("latitude") && hash.key?("longitude")
+        @lat_longs << [hash["longitude"], hash["latitude"]]
+      else 
+        puts "The latitude and longitude are not present"
```
Comment (hpjaj) [inline app/controllers/code_schools_controller.rb]: Why the `puts` statements here and on line 15?  Is that for your debugging an issue?  

If the goal is for the content in those lines to persist in some way, you'll want to use a `Rails.logger.info "some message content"` in place of those lines.  That way, the content will print out to the server logs.  

Or alternatively, is your intent to provide the user with feedback?  If yes, you'd want to use a `flash[:error]` message.

That said, was is the goal here?

File: app/views/code_schools/map.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<h1>Code Schools Map</h1>
```
Comment (hpjaj) [inline app/views/code_schools/map.html]: Curious, why is this file a `.html` file type vs. a `.html.erb` file type?

File: app/views/veterans/map.html [hunk 3]
```diff
@@ -2,3 +2,4 @@
 
 <%= content_tag 'div', nil, id: :'veterans-map', class: :'veterans-map', data: { lat_longs: @lat_longs } %>
```
Comment (hpjaj) [inline app/views/veterans/map.html]: This file type, too?

File: app/controllers/code_schools_controller.rb (hunk 1 above)
Comment (hd719) [inline app/controllers/code_schools_controller.rb]: Sorry it was for debugging purposes, will fix it asap!

File: app/views/code_schools/map.html (hunk 2 above)
Comment (hd719) [inline app/views/code_schools/map.html]: I honestly don't know I think it came like that, but I can have it changed to `.html.erb`.

File: app/views/veterans/map.html (hunk 3 above)
Comment (hd719) [inline app/views/veterans/map.html]: This one I made so I can def. change it to `.html.erb`.

**Comment by hd719** [comment]
@hpjaj @rickr I have made changes from above please let me know what you think. If anything is wrong I will def. fix it asap. 

-Thank you.

**Comment by hpjaj** [comment]
@hd719 - I am just a member of the Operation Code community, I don't have reviewer permissions, for merging.  Was just doing a some code reviews to provide feedback.

Ironically, though, I also went to Bloc.io 🙂

**Comment by hd719** [comment]
@hpjaj It is always a pleasure to meet a fellow Bloc alumni!

File: app/controllers/code_schools_controller.rb [hunk 4]
```diff
@@ -7,4 +7,16 @@ def index
     @code_schools_by_state = code_schools.by_state
     @country = Country.named('United States')
   end
+
+  def map
+    code_schools = CodeSchools.new.all
+    @lat_longs = []
+
+    code_schools.each do |hash|
```
Comment (rickr) [inline app/controllers/code_schools_controller.rb]: I know it's a limited use variable but I think `school` would be more descriptive here than `hash`.

Comment (rickr) [inline app/controllers/code_schools_controller.rb]: Would it make sense to move this to the `CodeSchools` class? Maybe something like `CodeSchools.lat_longs`? I think it would make things easier to test and it would clean up the controller a bit.

File: config/routes.rb [hunk 5]
```diff
@@ -18,16 +18,18 @@
 
   # LetsEncrypt
   get '/.well-known/acme-challenge/:id' => 'pages#letsencrypt'
-
   get '/veterans/thanks', to: 'veterans#thanks'
-  get '/veterans/map', to: 'veterans#map'
```
Comment (rickr) [inline config/routes.rb]: I'm OK with changing the URL but I think we should maintain the old one (or at least 301 to the new URL) in case anyone has linked to it.

File: config/routes.rb [hunk 6]
```diff
@@ -18,16 +18,18 @@
...
   get '/.well-known/acme-challenge/:id' => 'pages#letsencrypt'
   resources :veterans, only: [:new, :create]
   post '/veterans/claim/:veteran', to: 'veterans#claim', as: :veterans_claim
   post '/veterans/unclaim/:veteran', to: 'veterans#unclaim', as: :veterans_unclaim
 
   resources :donations, only: [:index, :new, :create]
 
+  # Maps
+  get '/maps/veterans', to: 'veterans#map'
```
Comment (rickr) [inline config/routes.rb]: If we're updating the URL does it make sense to create a new `MapsController`? It seems like a lot of the code is similar so maybe it would clean up some duplication as well.

**Comment by rickr** [comment]
It also looks like codeclimate picked up a few new issues. Can you please address those as well?

File: app/controllers/code_schools_controller.rb (hunk 4 above)
Comment (hd719) [inline app/controllers/code_schools_controller.rb]: Hey @rickr when you mean `move this to the CodeSchools` I'm sorry, but what are you referring too?

Comment (rickr) [inline app/controllers/code_schools_controller.rb]: It looks like this is pulling out the lat/longs for each school. I was wondering if it makes more sense to have this in the `CodeSchools` class as something like `CodeSchools.with_lat_longs` or something like that.

**Comment by hd719** [comment]
Hey @rickr I moved some of the logic from the `code_school_controller.rb` to the code school's class. 

Let me know what you think. Also the codeclimate tests how do you fix the trailing whitespace? 

-Thank you.

**Comment by hd719** [comment]
Hey @rickr any update on this?

File: app/views/veterans/map.html.erb [hunk 7]
```diff
@@ -2,3 +2,4 @@
 
 <%= content_tag 'div', nil, id: :'veterans-map', class: :'veterans-map', data: { lat_longs: @lat_longs } %>
 
+<script>console.log(document.getElementById("veterans-map").dataset["lat-longs"]);</script>
```
Comment (rickr) [inline app/views/veterans/map.html.erb]: Can you remove this logging statement?

File: app/controllers/code_schools_controller.rb [hunk 8]
```diff
@@ -7,4 +7,8 @@ def index
     @code_schools_by_state = code_schools.by_state
     @country = Country.named('United States')
   end
-end
+
+  def map
+    @code_schools = CodeSchools.new.lat_longs
```
Comment (rickr) [inline app/controllers/code_schools_controller.rb]: This name is kind of confusing when looking at it in the view. Can you rename this to either @code_schools_lat_longs, or just @lat_longs?

**Comment by hd719** [comment]
@rickr I will fix these changes asap. 

- Thank you.

**Comment by rickr** [comment]
Hey @hd719 - it looks like you're all done with the latest fixes - is that the case?

**Comment by hd719** [comment]
@rickr Hey! so yeah I implemented the latest fixes however for some reasonI cannot get the codeclimate tests to pass.

**Comment by rickr** [comment]
It looks like a lot of them are little space things:
For example this file is missing a new line: https://github.com/OperationCode/operationcode/pull/737/files#diff-5665789a04effed313d97af65082f8feR33

**Comment by hd719** [comment]
Hey @rickr so for the example did I add the new line correctly? 

I think something is wrong with my editor

**Comment by hd719** [comment]
Hey @rickr! So the codeclimate issues have been fixed, there is this 1 issue about the content tag, but that was already there from the beginning and I don't know how to fix that. 

- Thank you.

**Comment by hollomancer** [comment]
You can definitely add yourself to contributors.md - looks like Codeclimate caught one more thing though.

**Comment by hd719** [comment]
Hey @hollomancer good to hear from you. 

It says `Unescaped model attribute in content_tag` 

```
<%= content_tag 'div', nil, id: :'veterans-map', class: :'veterans-map', data: { lat_longs: @lat_longs } %>
```

I do not know how to fix this error. Any help would def. be appreciated. 

-Thank you.

**Comment by rickr** [comment]
I think the finding might be bogus. 

I believe content_tag marks input as `html_safe` which in rails means you are sure the content is not malicious. Since we're reviewing the input in a PR I've gone ahead and approved the change anyway.

**Comment by hd719** [comment]
@hollomancer and @rickr Thank you! It was fun working on this PR glad I could be of help!

**Comment by rickr** [comment]
Great job. Thanks for all your hard work!
